### PR TITLE
fix: disable default save page as

### LINKED ISSUE #1541: feat: disable default "save page as" keyboard shortcut
## Description

Currently, pressing `cmd+s` / `ctrl+s` when writing in the block editor will open the browser default "save page as" window to save the current HTML page to a local file.

People tired of old office software may frequently press this shortcut to avoid loss of work progress. And edition has been synced with yjs so that there is no concept of "saving" here?

## Solution

I think we can disable this shortcut to avoid interrupting the writing flow.

### PR DESCRIPTION
**Pull request by soumyasen10**
Fixes #1541

### Currently, pressing cmd+s / ctrl+s when writing in the block editor will not  open  the browser default "save page as" window to save the current HTML page to a local file,
**It's open a toast which shows a toast "Save" at the bottom right corner.**
2024-04-14 12_43_26-Settings

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -173,5 +173,8 @@
     "macos-alias": "npm:@napi-rs/macos-alias@0.0.4",
     "fs-xattr": "npm:@napi-rs/xattr@latest",
     "@radix-ui/react-dialog": "npm:@radix-ui/react-dialog@latest"
+  },
+  "dependencies": {
+    "react-hot-toast": "^2.4.1"
```
Comment (pengx17) [inline package.json]: please do not introduce another toast library. we already have one in `@affine/component`

File: packages/frontend/templates/templates.gen.ts [hunk 2]
```diff
@@ -1,11 +1,11 @@
 /* eslint-disable simple-import-sort/imports */
 // Auto generated, do not edit manually
-import json_0 from './onboarding/info.json';
-import json_1 from './onboarding/blob.json';
-import json_2 from './onboarding/W-d9_llZ6rE-qoTiHKTk4.snapshot.json';
+import json_0 from './onboarding/W-d9_llZ6rE-qoTiHKTk4.snapshot.json';
```
Comment (pengx17) [inline packages/frontend/templates/templates.gen.ts]: please don't commit this file

File: packages/frontend/web/src/app.tsx [hunk 3]
```diff
@@ -66,13 +67,37 @@ export function App() {
...
+    try {
+      document.addEventListener('keydown', handleKeyDown, false);
+    } catch (error) {
+      console.error('Error attaching event listener:', error);
+    }
+
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown, false);
+    };
+  }, []);
```
Comment (pengx17) [inline packages/frontend/web/src/app.tsx]: Instead adding a new useEffect hook, you can follow https://github.com/toeverything/AFFiNE/blob/9b620ecbc94e40f49706f22c912d11a4607911da/packages/frontend/core/src/modules/navigation/view/use-register-navigation-commands.ts#L30-L45 and register a hidden command to ignoring incoming `$mod+s` keybindings

**Review by pengx17 — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/app.tsx (hunk 3 above)
Comment (soumyasen10) [reply packages/frontend/web/src/app.tsx]: okay i will fix it.

Comment (soumyasen10) [reply packages/frontend/web/src/app.tsx]: Fixes: #6548
@pengx17 **i am resolve all the issues you are asked for. please kindly check my new pr. thanks**

File: packages/frontend/core/src/modules/navigation/view/use-register-navigation-commands.ts [hunk 4]
```diff
@@ -43,6 +44,22 @@ export function useRegisterNavigationCommands() {
...
+        keyBinding: {
+          binding: '$mod+s',
+        },
+        label: 'Ignore Mod+S',
+        icon: null,
+        run() {
+          toast('save');
+        },
+      })
+    );
```
Comment (pengx17) [inline packages/frontend/core/src/modules/navigation/view/use-register-navigation-commands.ts]:
```suggestion
    unsubs.push(
      registerAffineCommand({
        id: 'alert-ctrl-s',
        category: 'affine:general',
        preconditionStrategy: PreconditionStrategy.Never,
        keyBinding: {
          binding: '$mod+s',
        },
        label: '',
        icon: null,
        run() {
          toast('save');
        },
      })
    );
```

File: packages/frontend/core/src/modules/navigation/view/use-register-navigation-commands.ts [hunk 5]
```diff
@@ -43,6 +44,22 @@ export function useRegisterNavigationCommands() {
...
+        id: 'alert-ctrl-s',
+        category: 'affine:general',
+        preconditionStrategy: PreconditionStrategy.Never,
+        keyBinding: {
+          binding: '$mod+s',
+        },
+        label: '',
+        icon: null,
+        run() {
+          toast('save');
```
Comment (pengx17) [inline packages/frontend/core/src/modules/navigation/view/use-register-navigation-commands.ts]: Please add i18n for `save`.

**Comment by pengx17** [comment]
There are merge conflicts.

File: packages/frontend/core/src/modules/navigation/view/use-register-navigation-commands.ts (hunk 5 above)
Comment (soumyasen10) [reply packages/frontend/core/src/modules/navigation/view/use-register-navigation-commands.ts]: i am resolve all the things please check #6548
@pengx17 
this is the final change pr b3de71bf7e067fb75f6cd5c617b4e9a4a71fded3

File: package.json [hunk 6]
```diff
@@ -78,7 +78,7 @@
     "@vitest/ui": "1.4.0",
     "cross-env": "^7.0.3",
     "electron": "^29.3.0",
-    "eslint": "^8.57.0",
+    "eslint": "^9.0.0",
```
Comment (darkskygit) [inline package.json]: please don't do multiple things in one PR

**Review by darkskygit — DISMISSED** [review]

**Review by pengx17 — APPROVED** [review]

**Comment by pengx17** [comment]
Thank you for your contribution. I have revised the PR to meet the quality standards of the repository.

**Pull request merged**